### PR TITLE
XW-422 throw exception when method called from non POST request

### PR DESCRIPTION
**Pull request by bognix**
https://wikia-inc.atlassian.net/browse/XW-422

PR in Mercury: https://github.com/Wikia/mercury/pull/1466

### CODE REVIEW

**Comment by vforge** [comment]
LGTM

File: extensions/wikia/CuratedContent/CuratedContentValidatorController.class.php [hunk 1]
```diff
@@ -15,6 +16,11 @@ public function __construct() {
 	}
 
 	public function validateSection() {
+		global $wgRequest;
+		if (!$wgRequest->wasPosted()) {
```
Comment (rogatty) [inline extensions/wikia/CuratedContent/CuratedContentValidatorController.class.php]: Missing spaces ;)

File: extensions/wikia/CuratedContent/CuratedContentValidatorController.class.php [hunk 2]
```diff
@@ -15,6 +16,11 @@ public function __construct() {
 	}
 
 	public function validateSection() {
+		global $wgRequest;
```
Comment (rogatty) [inline extensions/wikia/CuratedContent/CuratedContentValidatorController.class.php:19]: Quoting https://github.com/Wikia/app/blob/dev/includes/wikia/nirvana/WikiaController.class.php#L143:

> if you want to access request, use: RequestContext::getMain()->getRequest()

File: extensions/wikia/CuratedContent/CuratedContentValidatorController.class.php (hunk 1 above)
Comment (vforge) [inline extensions/wikia/CuratedContent/CuratedContentValidatorController.class.php]: @Warkot

File: extensions/wikia/CuratedContent/CuratedContentValidatorController.class.php (hunk 2 above)
Comment (bognix) [inline extensions/wikia/CuratedContent/CuratedContentValidatorController.class.php:19]: above comment makes sense in context of method it applies to. In this particular case I don't see difference

Comment (rogatty) [inline extensions/wikia/CuratedContent/CuratedContentValidatorController.class.php:19]: Well, the difference is that we don't use a global. I don't know if it's better ;) PHP magician needed.

**Comment by rogatty** [comment]
LGTM

File: extensions/wikia/CuratedContent/CuratedContentValidatorController.class.php (hunk 2 above)
Comment (Warkot) [inline extensions/wikia/CuratedContent/CuratedContentValidatorController.class.php:19]: Lately I bumped into this https://github.com/Wikia/app/blob/XW-396/skins/oasis/modules/BodyController.class.php#L121, where `global $wgRequest` was exchanged for `F::app()->wg->Request`. Not using globals sounds right, but I'm just a low level PHP warlock. ;)

**Pull request merged**